plugins/org.eclipse.n4js/src/org/eclipse/n4js/typesystem/n4js.xsemantics

```diff
@@ -2007,33 +2014,21 @@ from {
 rule expectedTypeInMultiplicativeExpression
 	G |- MultiplicativeExpression e |> Expression expression : TypeRef T
 from {
-	if (jsVariantHelper.isTypeAware(e)) {
-		T = G.numberTypeRef
-	} else {
-		T = G.anyTypeRef
-	}
+	T = G.anyTypeRef
 }
 
 /* @see [N4JS] 6.1.14 Additive Expression */
 rule expectedTypeInAdditiveExpression
 	G |- AdditiveExpression e |> Expression expression : TypeRef T
 from {
-	if (e.op != AdditiveOperator.ADD && jsVariantHelper.isTypeAware(e)) {
-		T = G.numberTypeRef
-	} else {
-		T = G.anyTypeRef
-	}
+	T = G.anyTypeRef
```

Comment:
type `symbol` is disallowed, e.g. the following throws an exception at runtime:
```
let sym: symbol = Symbol.iterator;
sym+5;
```

Reply:
I added

```
let sym: symbol = Symbol.iterator;
// XPECT errors --> "Operand of type symbol cannot be converted to number." at "sym"
// XPECT type of n+sym --> union{number,string}
n+sym
// XPECT errors --> "Operand of type symbol cannot be converted to number." at "sym"
// XPECT type of sym+n --> union{number,string}
sym+n
// XPECT errors --> "Operand of type symbol cannot be converted to number." at "sym"
// XPECT type of s+sym --> string
s+sym
// XPECT errors --> "Operand of type symbol cannot be converted to number." at "sym"
// XPECT type of sym+s --> string
sym+s
```
(and similar for other operations, in these cases always expecting number).

I also improved the type inference of "+": I will never return "unknown type" but union{number,string} instead.

Reply:
> I also improved the type inference of "+": I will never return "unknown type" but union{number,string} instead.

Not sure about this one, because iirc we had the rule if an inference that depends on an unknown type should result to "unknown type" (i.e. don't suppress "unknown type" but pass on though hierarchy). But not sure right now. Would also be ok to keep as is.